lua/spectre_common.lua

```diff
@@ -67,6 +73,16 @@ local function has_marker_headers(headers, marker_header_list)
     return false
 end
 
+-- Encodes the id fields in request body as single string
+local function get_id_from_req_body(id_field, request_body)
+    local body = json:decode(request_body)
+    if body[id_field] ~= nil then
+        return tostring(body[id_field])
+    else
+        error('Id field not available in request body:' .. id_field)
```

Comment:
Can you add a test for this error case?

Reply:
Update unittest to test for error condition.